modules/nf-core/bowtie/align/main.nf

```diff
@@ -3,9 +3,7 @@ process BOWTIE_ALIGN {
     label 'process_high'
 
     conda "${moduleDir}/environment.yml"
-    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'https://depot.galaxyproject.org/singularity/mulled-v2-ffbf83a6b0ab6ec567a336cf349b80637135bca3:c84c7c55c45af231883d9ff4fe706ac44c479c36-0' :
-        'biocontainers/mulled-v2-ffbf83a6b0ab6ec567a336cf349b80637135bca3:c84c7c55c45af231883d9ff4fe706ac44c479c36-0' }"
+    container "nf-core/modules/bowtie_align:bowtie_align--d797c0ccd2d330fd"
```

Comment:
WHAT IS THIS MAGIC?

Reply:
Please mind that it is compulsory for all magicians to deposit samples of all their spells in the grimoire "_Magical Modules Mastery_" 🪄. 

[The tome can be found in the tool's repository,](https://github.com/nf-core/tools/tree/master/tests/data/mock_module_containers/modules) and the [corresponding wizardry ordeal is codified in over here](https://github.com/nf-core/tools/blob/master/tests/test_download.py#L185-L239).

Reply:
> Please mind that it is compulsory for all magicians to deposit samples of all their spells in the grimoire "Magical Modules Mastery" 🪄.

Didn't even know that was a thing 🙃

Reply:
Few people do, and luckily the container definitions in modules don't change frequently either. 

Whether it's a significant alteration or a minor linting oversight like not using the correct quote characters, the impact might unfortunately not be apparent until much later. Unless tested right away with `nf-core download`, it will take weeks to months until the first pipeline is released that uses an updated module. Failures due to a change in June 2023, for example, [were just noticed recently](https://github.com/nf-core/tools/pull/2768). 

Once an issue is identified, the problematic code has already been released in a pipeline revision and can't be retracted for fixing a quirky module. Therefore, the only viable option is to adapt `nf-core download` accordingly, which is cumbersome and sometimes not even possible any more, [because two pipelines require mutually exclusive adaptions](https://github.com/nf-core/tools/issues/2392#issuecomment-1682393518). Ironically, it was exactly an unconventional Bowtie module, that broke the downloadability for particular [chipseq](https://github.com/nf-core/chipseq/issues/318) and [atacseq](https://github.com/nf-core/atacseq/issues/270) revisions beyond repair.

The ideal solution would be a CI test running right away in the modules repo. However, I don't know if that is feasible, because individual Python functions from the tools package would need to be applied selectively to a module.